testcases/cve/cve-2017-17052.c

```diff
@@ -101,6 +103,7 @@ static void run(void)
 			usleep(EXEC_USEC);
 			*do_exit = 1;
 		}
+		SAFE_WAIT(&status);
```

Comment:
Library will reap children it finds. If you want to reap in test, then please check also expected return code.
See also: tst_reap_children() in docs.

Reply:
The test forks a child sleeps some time and then sets a shared int (*do_exit) to inform the forked child it should exit. This is done in a loop 4 times. As the shared int is reused for the following runs we have to wait for the process to actually exit before continuing and resetting *do_exit.
So this wait there is just for synchronizing with the child.

But I've now added checks to verify the exit status 0.

Reply:
Right, without wait some children would run longer than intended.